<launch>
<node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
    <!-- camera node -->
  <include file="$(find usb_cam)/launch/usb_cam-test.launch">
  </include>
    <!-- detect aruco markers -->
    <node pkg="nodelet" clear_params="true" type="nodelet" name="aruco_detect" args="load aruco_pose/aruco_detect nodelet_manager">
        <remap from="image_raw" to="/usb_cam/image_raw"/>
        <remap from="camera_info" to="/usb_cam/camera_info"/>
        <param name="length" value="0.7"/>
    </node>

   <node name = "cupDetectWithFunction" pkg = "cupDetection" type="cupDetectWithFunction" output = "screen">
      <param  name="XPose" type="int" value="10"/>
      <param  name="YPose" type="int" value="490"/>
      <param name = "cupDistance" type = "int" value = "24"/>
      
      <!-- 10 490 24-->  <!-- 447 520  28-->
   </node>	
   <!--<node name = "cupDetectWithFunction" pkg = "cupDetection" type="cupDetectWithFunction" output = "screen">
      <param  name="XPose" type="int" value="603"/>
      <param  name="YPose" type="int" value="465"/>
      <param name = "cupDistance" type = "int" value = "28"/>
   </node> -->	   
   <node name = "CameraResultWithoutMachineStatus" pkg="aruco_pose" type = "CameraResultWithoutMachineStatus" output = "screen">
     <param name="testStart" type = "int" value = "0"/>
     <param name = "angleMargin" type = "int" value = "30"/>
   </node>
    <!--  <node name = "CameraResult" pkg="aruco_pose" type = "CameraResult" output = "screen">
     <param name="testStart" type = "int" value = "0"/>
     <param name = "angleMargin" type = "int" value = "30"/>
   </node>
    -->
    <node name = "gui" pkg="eurobot_gui"  type="demo_guiForCamera.py"/>
</launch>
